[PL-Doc] Move project across organizations (#11619)
harness/developer-hub: docs/platform/organizations-and-projects/move-projects/overview.md

---
 title: Move a project 
 description: Overview of moving a project between organizations, including prerequisites, Supported modules and how to request or perform a move.
 sidebar_label: Overview
 sidebar_position: 1
 tags:
   - organizations
   - projects
 keywords:
   - move projects
   - move project between organizations
   - transfer project
   - organization restructure
---

> The move projects feature allows you to transfer projects between organizations, which is useful during reorganizations, or ownership transitions.

Reviewer: Use singular

> This capability enables teams to adapt to structural changes while preserving project configuration, resources, and history, supporting flexible governance across evolving team structures.

Reviewer: Remove this

> A project is a group of harness modules, their pipelines, and other resources used to organize and manage resources in an organization. The move projects feature allows you to transfer projects between organizations, which is useful during reorganizations, or ownership transitions.

Reviewer: ```suggestion
The move projects feature allows you to transfer projects between organizations, which is useful during reorganizations, or ownership transitions.
```

This capability enables teams to adapt to structural changes while preserving project configuration, resources, and history, supporting flexible governance across evolving team structures.

## What happens when you move a project 

When you move a project to another organization in Harness, several actions take place in the background.

> 1. **Entities inside the project are moved:** Pipelines, services, environments, and other [entities](#supported-entities) are carried over to the new organization. These entities remain intact; however, references to external items are validated during the move.

Reviewer: This is not entirely true. We should point here to the doc that talks about the supported modules/entities.

> 4. **Audit logs**: Old audit logs are maintained, but redirect links to the source organization may break.

Reviewer: Links are not to the organization. Mention clearly what will break and why

For detailed pre-move verification and post-move remediation steps, refer to the [project movement checklist](project-movement-checklist.md).

## Prerequisites and permissions

> To move a project across organizations, you need the following roles on the project, its source organization, and the destination organization:

Reviewer: 1. These are not roles but permissions
2. " on the project, its source organization, and the destination organization:" -> Remove this as permissions are for specific resource not for all
3. Permission should be replaced with permission
4. "Create Permission on the destination organization." => Create project permission in the destination organization
5. "Move Permission on the..... " -> "Move Permission for the"

- Move Permission on the project that you want to move.
- Create Permission on the destination organization.

## Supported modules

* [Platform](/docs/platform/platform-whats-supported)
* [Continuous Delivery](/docs/continuous-delivery/cd-integrations)
* [Continuous Integration](/docs/continuous-integration/use-ci/set-up-build-infrastructure/which-build-infrastructure-is-right-for-me/#feature-compatibility-matrix)
* [Internal Developer Portal](/docs/internal-developer-portal/whats-supported)
* [Security Test Orchestration](/docs/security-testing-orchestration/whats-supported/sto-deployments)
* [Supply Chain Security](/docs/software-supply-chain-assurance/ssca-supported)
* [Code Repository](/docs/code-repository/code-supported)
* [Database DevOps](/docs/database-devops/dbdevops-supported-platforms.md)
* [Software Engineering Insights](/docs/software-engineering-insights/sei-support)

> :::note [Unsupported Entities](/docs/platform/references/harness-entity-reference)

Reviewer: For Code module, we will add the redirection unsupported point

---
 title: Move a project 
 description: Overview of moving a project between organizations, including prerequisites, Supported modules and how to request or perform a move.
 sidebar_label: Overview
 sidebar_position: 1
 tags:
   - organizations
   - projects
   - move project
 keywords:
   - move projects
   - move project between organizations
   - transfer project
   - organization restructure
---

:::note Feature availability
This feature requires the `PL_PROJECT_MOVEMENT_ENABLED` feature flag. Contact [Harness support](mailto:support@harness.io) to enable it.
:::

> This feature allows you to move a project across organizations within your Harness account. This capability is essential for organizational restructuring, team reorganizations, or when consolidating projects under different business units.

Reviewer: ```suggestion
This feature allows you to transfer a project from one organization to another within your Harness account. It can be used when a project needs to be moved to a different organization, such as during ownership transfers.
```

## What happens when you move a project 

1. **Entities are moved**: [Supported entities](#supported-modules), such as pipelines, services, environments, and monitored services, are carried over with the project to the destination organization.

> 2. **Some references need to be reconfigured**: Resources scoped at the source organization level —such as connectors, secrets, templates, webhooks, or notifications—may become inaccessible after the move. These references need to be recreated or updated in the destination organization. References to account- or project-level entities remain intact.

Reviewer 2: > Some references need to be reconfigured

lets avoid using vague word like some.
suggestion to use org level references need to be reconfigured

> 2. **Some references need to be reconfigured**: Resources scoped at the source organization level —such as connectors, secrets, templates, webhooks, or notifications—may become inaccessible after the move. These references need to be recreated or updated in the destination organization. References to account- or project-level entities remain intact.

Author: addresed.

> 2. **Organization-scoped resources need reconfiguration**: Resources scoped at the source organization level become inaccessible after the move. You must recreate connectors, secrets, templates, webhooks, and notifications in the destination organization, then update any project references to point to these new resources. References to account-level or project-level entities remain intact.

Author: resolved. - "You might need to recreate connectors, secrets, templates, webhooks, and notifications in the destination organization, and then update any project references to point to these new resources."

> 3. **Access control and policies**: Organization-level RBAC policies, governance rules, and security policies do not transfer and must be reapplied in the destination organization. When a project move is initiated, all access control components are migrated asynchronously. During this process, access to the project will be temporarily blocked for users.

Reviewer: ```suggestion
3. **Access control and policies**: Organization-level RBAC policies, governance rules, and security policies do not transfer and must be reapplied in the destination organization. When a project move is initiated, all access control components belonging to the project are migrated asynchronously. During this process, access to the project will be temporarily blocked for other users.
```

> 4. **Audit logs**: Existing audit logs are preserved, but any links in the logs pointing to source organization resources may break, as those resources will no longer be accessible.

Reviewer 2: > resources may break
will not be accessible.
Existing audit logs are preserved, but any reference links in the audits pointing to a resource belonging to organization/project resources will not be accessible, as those resources will no longer be accessible.

give an example

> 3. **Access control requires reconfiguration**: Organization-level RBAC policies, governance rules, and security policies do not transfer and must be recreated in the destination organization. All Project-level access control components including role bindings, resource groups, and roles are migrated asynchronously. While the move is in progress, users may experience temporary access restrictions during the move process.

Reviewer: Do not use "must". It depends on if this would be really reuired.

> 4. **Audit logs**: Existing audit logs remain in the source organization and are not moved with the project. Any links in these logs pointing to the moved project or older organization will become inaccessible.

Reviewer: Split it into two items:
1. Audit logs at Account level: Everything should work fine. (Confirm with @arya-harness once)
2. Audit logs at Organization level: Logs belonging to the project before the project is moved will not be moved along with the project to the destination organization. Any links in these logs pointing to the moved project in the older organization will break.

Reviewer 2: 1. correct.
2. correct, to maybe add onto this section, we can also define that, any new audits belonging to the moved project show up in the newer organization.

For detailed pre-move validation and post-move remediation steps, see the [Move a Project guide](./pre-move-and-post-move-guide.md).

## Prerequisites

To move a project across organizations, you need the following permissions:

- Move permission for the project that you want to move.
- Create project permission in the destination organization.

## Supported modules

* [Platform](/docs/platform/platform-whats-supported)
* [Continuous Delivery and GitOps](/docs/continuous-delivery/cd-integrations)
* [Continuous Integration](/docs/continuous-integration/use-ci/set-up-build-infrastructure/which-build-infrastructure-is-right-for-me/#feature-compatibility-matrix)
* [Internal Developer Portal](/docs/internal-developer-portal/whats-supported)
* [Security Test Orchestration](/docs/security-testing-orchestration/whats-supported/sto-deployments)

> * [Supply Chain Security](/docs/software-supply-chain-assurance/ssca-supported)

Author: Ack removed.

 sidebar_position: 1
 tags:
   - organizations
   - projects
   - move project
 keywords:
   - move projects
   - move project between organizations
   - transfer project
   - organization restructure
---

:::note Feature availability
This feature requires the `PL_PROJECT_MOVEMENT_ENABLED` feature flag. Contact [Harness support](mailto:support@harness.io) to enable it.
:::

This feature allows you to transfer a project from one organization to another within your Harness account. It can be used when a project needs to be moved to a different organization, such as during ownership transfers.

## What happens when you move a project 

1. **Entities are moved**: [Supported entities](#supported-modules), such as pipelines, services, environments, and monitored services, are carried over with the project to the destination organization.

2. **Organization-scoped resources need reconfiguration**: Resources scoped at the source organization level become inaccessible after the move. You might need to recreate connectors, secrets, templates, webhooks, and notifications in the destination organization, and then update any project references to point to these new resources.

> 3. **Access control requires reconfiguration**: Organization-level RBAC policies, governance rules, and security policies do not transfer and must be recreated in the destination organization. All Project-level access control components including users, service accounts, user groups, role bindings, resource groups, and roles are moved asynchronously. While the move is in progress, users or service accounts that have access to the project might temporarily be unable to access it.

Reviewer: 1. What is meant by "RBAC policies, governance rules, and security policies"? I have flagged this in the past as well. 
2. We cannot use "reuires/need configuration". Even this was highlighted in the past.

4. **Audit logs**: 

>     - Account-level audit logs: No impact - all logs remain accessible.

Reviewer: We do not need to mention account level audit logs.
